### PR TITLE
Bootstrap release process

### PR DESCRIPTION
**Pull request by nikimanoledaki**
**What this PR does / why we need it**:

This PR bootstraps a release process based on [GoReleaser](https://goreleaser.com/).

**Which issue(s) this PR fixes**:
Fixes part of https://github.com/open-component-model/ocm/issues/101

### Release Process steps
The steps required to do a release, as introduced by this PR, are the following:

- Create a PR with the following:
	- Update the version in `pkg/version/release.go`.
	- Add release notes in `docs/releasenotes` for a given version. e.g. for `v0.2.0`, create `docs/releasenotes/v0.2.0.md`. We can automate drafting release notes in a later PR.
- Request 2 reviews & approvals to merge the PR.
- Once the PR is merged, start the `Release` GitHub Action. This will run the tests and linter, check that the release notes are present for this version, create a tag, and trigger a release through goreleaser.

This process is different from release processes that are triggered by pushing a tag. It is based on the eksctl release process. By opening a PR, we can ensure, for example, that release notes are added and reviewed (so that nothing is missing), and these can be reviewed and approved by other maintainers. This way, this release process is more GitOps-based than tag-based release processes.

This work was tested E2E in a personal repo.

~## Next Steps~
~I started refactoring the script based on feedback from @phoban01 and @souleb to move everything from the script into the GitHub Action workflow as separate steps. I'm working on this, it is still a WIP. I can add these changes in a separate PR once I get them working. I opened this PR without these refactorings to get this reviewed and merged asap.~

The conditional statement in `pkg/version/generate/release_generate.go` can be expanded to add logic for formatting prereleases as well. See eksctl for an example of possible next steps: https://github.com/weaveworks/eksctl/blob/main/pkg/version/generate/release_generate.go

### CODE REVIEW

**Comment by nikimanoledaki** [comment]
Update: Moved everything from the script to the workflow so this is all self-contained and no longer uses a `make` command either :)

File: pkg/version/generate/release_generate.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/open-component-model/ocm/pkg/version"
```
Comment (Skarlso) [inline pkg/version/generate/release_generate.go:13]:
```suggestion
	"github.com/Masterminds/semver/v3"

	"github.com/open-component-model/ocm/pkg/version"
```

Comment (Skarlso) [reply pkg/version/generate/release_generate.go:13]: This should fix the linter.

**Review by Skarlso — APPROVED** [review]
LGTM

**Review by phoban01 — APPROVED** [review]
LGTM

**Pull request merged**